### ISSUE TITLE
Failed to upload/download/generate file: [Error : 0 : Too big file with unknown size]

### ISSUE THREAD

**Issue by JunaidBabu** [tdlib/td#422]
I'm unable to download certain files.
I run getRemoteFile(remote_id) to get the file_id.
Then I run downloadFile(file_id).
The download proceeds till some 20% and it freezes there. I retried the above steps after deleting the partially downloaded file, it again stopped at 20%.
Error logs showed the below:

```
148 ^[[0m^[[1;33m[ 2][t 4][1544323064.184988737][FileManager.cpp:2429][!FileManager]        Failed to upload/download/generate file: [Error : 0 : Too big file with unknown size].     Query type = 2. File type is documents
149 ^[[0m^[[1;33m[ 2][t 3][1544323064.402849436][Session.cpp:404][!Session:1:download_small#0]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#0::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:41914]
150 ^[[0m^[[1;33m[ 2][t 3][1544323064.467171907][Session.cpp:404][!Session:1:download_small#1]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#1::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:41916]
151 ^[[0m^[[1;33m[ 2][t 4][1544323164.890347958][FileManager.cpp:2429][!FileManager]        Failed to upload/download/generate file: [Error : 0 : Too big file with unknown size].     Query type = 2. File type is documents
152 ^[[0m^[[1;33m[ 2][t 3][1544323165.126140594][Session.cpp:404][!Session:1:download_small#1]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#1::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:41950]
153 ^[[0m^[[1;33m[ 2][t 3][1544323165.176815271][Session.cpp:404][!Session:1:download_small#0]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#0::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:41948]
154 ^[[0m^[[1;33m[ 2][t 4][1544323268.095056772][FileManager.cpp:2429][!FileManager]        Failed to upload/download/generate file: [Error : 0 : Too big file with unknown size].     Query type = 2. File type is documents
155 ^[[0m^[[1;33m[ 2][t 3][1544323268.315089941][Session.cpp:404][!Session:1:download_small#0]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#0::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:41984]
156 ^[[0m^[[1;33m[ 2][t 3][1544323268.371797562][Session.cpp:404][!Session:1:download_small#1]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#1::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:41982]
157 ^[[0m^[[1;33m[ 2][t 4][1544323370.388631582][FileManager.cpp:2429][!FileManager]        Failed to upload/download/generate file: [Error : 0 : Too big file with unknown size].     Query type = 2. File type is documents
158 ^[[0m^[[1;33m[ 2][t 3][1544323370.583730698][Session.cpp:404][!Session:1:download_small#1]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#1::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:42018]
159 ^[[0m^[[1;33m[ 2][t 3][1544323370.661969185][Session.cpp:404][!Session:1:download_small#0]      on_closed: [Error : 2 : Too much dropped packets [total_size:256KB]] Session:1:    download_small#0::Connect::Tcp::[149.154.175.50:443] to DcId{1} from [192.168.0.17:42016]
160 ^[[0m
```

**Comment by levlam** [comment]
Currently, downloding of files bigger than 96000 kB only by remote_id is unsupported. We need to know a message in which the file was sent to be able to download it.

If you need a temporary workaround, then you can replace in the line https://github.com/tdlib/td/blob/101aa73f132d5124d8da45bae11a29c423420072/td/telegram/files/PartsManager.cpp#L41 32 with 512. As a side effect all files will be downloaded by parts of size 512 kB and there will be at most 4 simultaneous download file part requests, so files download will be slower.

**Comment by levlam** [comment]
This was fixed in TDLib 1.5.0, but to be able to download a file via remote ID, the corresponding message is required to be preloaded and accessible nevertheless.
